src/utils/helpers.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * prepends first i characters and appends last
+ * j characters of the string input to an ellipse.
+ **/
+const condenseText = (input, prefixLen, suffixLen) => {
+  let result = input;
+
+  if (typeof input !== 'string') {
+    result = JSON.stringify(input);
+  }
+
```

Comment:
This function may feel more natural if there wasn't a magic number `12` in it that heavily impacts the way that it functions, do you think that the user could check the length outside of the `condenseText` function before passing it in? The function in its current form seems more like `maybeCondenseText`

Reply:
There is also no assert that the length of the `input` is strictly greater than the `prefixLen + suffixLen`

Reply:
Yeah, I agree. Maybe rather than hardcoding `12`, if length of `input` isn't greater than `prefix` + `suffix` then it should just return the result (as you have it now).

Reply:
I added an optional condense threshold parameter such that data can grow and start condensing at some threshold of size.

Reply:
Also added the assertion for prefix + suffix